src/bin/test.rs

```diff
@@ -1,292 +1,358 @@
+                } else if args[0] == "copy_from_process" {
+                    let stack_var: usize = args[1].parse().unwrap();
+                    let heap_var: usize = args[2].parse().unwrap();
+                    test_copy_from_process(stack_var, heap_var)
+                } else {
+                    Err(format!("Len 3: Unknown test option: {}", args[0]).into())
+                }
             }
-            "spawn_name_wait" => {
-                let num_of_threads: usize = args[1].parse().unwrap();
-                spawn_name_wait(num_of_threads)
+            4 => {
+                if args[0] == "merged_mappings" {
+                    let path = &args[1];
+                    let mapped_mem: usize = args[2].parse().unwrap();
+                    let mem_size: usize = args[3].parse().unwrap();
+                    test_merged_mappings(path.to_string(), mapped_mem, mem_size)
+                } else {
+                    Err(format!("Len 4: Unknown test option: {}", args[0]).into())
+                }
             }
-            _ => Err(format!("Len 2: Unknown test option: {}", args[0]).into()),
+            _ => Err("Unknown test option".into()),
+        }
+    }
+}
+
+#[cfg(target_os = "windows")]
+mod windows {
+    use super::*;
+    use std::mem;
+    use windows_sys::Win32::{
+        Foundation::CloseHandle,
+        System::{
+            Diagnostics::Debug::{GetThreadContext, CONTEXT, EXCEPTION_POINTERS, EXCEPTION_RECORD},
+            Threading::{GetCurrentProcessId, GetCurrentThread, GetCurrentThreadId},
         },
-        3 => {
-            if args[0] == "find_mappings" {
-                let addr1: usize = args[1].parse().unwrap();
-                let addr2: usize = args[2].parse().unwrap();
-                test_find_mappings(addr1, addr2)
-            } else if args[0] == "copy_from_process" {
-                let stack_var: usize = args[1].parse().unwrap();
-                let heap_var: usize = args[2].parse().unwrap();
-                test_copy_from_process(stack_var, heap_var)
-            } else {
-                Err(format!("Len 3: Unknown test option: {}", args[0]).into())
+    };
+
+    #[inline(never)]
+    pub(super) fn real_main(args: Vec<String>) -> Result<()> {
+        let exception_code = u32::from_str_radix(&args[0], 16).unwrap() as i32;
+
+        // Generate the exception and communicate back where the exception pointers
+        // are
+        unsafe {
+            let mut exception_record: EXCEPTION_RECORD = mem::zeroed();
+            let mut exception_context: CONTEXT = mem::zeroed();
+            exception_context.ContextFlags =
+                minidump_common::format::ContextFlagsAmd64::CONTEXT_AMD64_ALL.bits();
+
+            let pid = GetCurrentProcessId();
+            let tid = GetCurrentThreadId();
+
+            let thread_handle = GetCurrentThread();
+            GetThreadContext(thread_handle, &mut exception_context);
+            CloseHandle(thread_handle);
```

Comment:
nit: You don't have to use `CloseHandle` on this because `GetCurrentThread()` returns a pseudohandle and MSDN guarantees that `CloseHandle` is a no-op.

So you could replace all 3 lines with this one-liner:

`GetThreadContext(GetCurrentThread(), &mut exception_context);`